### PR TITLE
Test app: Fix the validation of getExternalProviders API

### PR DESCRIPTION
**Pull request by anmaur**
- Fix the validation of getExternalProviders API
- Extend the ApiWithCheckBoxInput to ingest setResult callback which allows callback based APIs to set the result of the text box.

### CODE REVIEW

File: apps/teams-test-app/src/components/privateApis/FilesAPIs.tsx [hunk 1]
```diff
@@ -133,13 +133,13 @@ const GetExternalProviders = (): React.ReactElement =>
     name: 'getExternalProviders',
     title: 'Get External Providers',
     label: 'excludeAddedProviders',
-    onClick: async (excludeAddedProviders: boolean) => {
+    onClick: async (excludeAddedProviders: boolean, setResult: (result: string) => void) => {
       let result;
```
Comment (MarcinJuraszek) [inline apps/teams-test-app/src/components/privateApis/FilesAPIs.tsx:137]: Is this really used anymore?

Comment (anmaur) [reply apps/teams-test-app/src/components/privateApis/FilesAPIs.tsx:137]: Yes, For the promise based apis which still uses callback

Comment (MarcinJuraszek) [reply apps/teams-test-app/src/components/privateApis/FilesAPIs.tsx:137]: I get that for the generic case, but in this particular one, you're doing `let result` and at the end `return result` without actually assigning anything to it.

File: apps/teams-test-app/src/components/privateApis/FilesAPIs.tsx [hunk 2]
```diff
@@ -133,13 +133,13 @@ const GetExternalProviders = (): React.ReactElement =>
...
       const callback = (error: SdkError, providers: files.IExternalProvider[]): void => {
         if (error) {
-          result = JSON.stringify(error);
+          setResult(JSON.stringify(error));
         } else {
-          result = JSON.stringify(providers);
+          setResult(JSON.stringify(providers));
         }
       };
       await files.getExternalProviders(excludeAddedProviders, callback);
```
[the remark this answers is not in the document]
Comment (anmaur) [reply apps/teams-test-app/src/components/privateApis/FilesAPIs.tsx]: makes sense. pushed a cleanup

**Review by jadahiya-MSFT — APPROVED** [review]

**Pull request merged**